### PR TITLE
Refactor: Add event handlers

### PR DESCRIPTION
**Pull request by jakubcolony**
While working on [colonyCDapp#267](https://github.com/JoinColony/colonyCDapp/issues/267), I came to a conclusion `eventProcessor` has grown quite large (close to 500 lines) and it made sense to apply the same handlers pattern I implemented for actions to other events.
Now the logic for handling each event is encapsulated within its own handler, and eventProcessor responsibility is solely to call the relevant handler.

Since there's much more nesting now, I thought it'd be beneficial to define paths in tsconfig, to have cleaner imports.

This PR is quite big, so I went through and marked the changes that are simply moving things around, without modifying the code. 

## Testing
Checkout https://github.com/JoinColony/colonyCDapp/pull/263 branch. Run block-ingestor separately from the rest of the CDapp env. Please check all events are still being picked up and handled as expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "rm -rf dist/* run/* && npx tsc",
     "start": "node dist/index.js",
-    "dev": "ts-node-dev src/index.ts",
+    "dev": "ts-node-dev -r tsconfig-paths/register src/index.ts",
```
Comment (jakubcolony) [inline package.json:9]: Needed in order for ts-node to pick up `paths` from tsconfig

File: src/handlers/actions/payment.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { utils } from 'ethers';
```
Comment (jakubcolony) [inline src/handlers/actions/payment.ts:1]: This handler is a bit messy, and I will clean it up when we start handling more actions and I can identify some refactoring opportunities

File: src/handlers/colonies/colonyAdded.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { colonySpecificEventsListener } from '~/eventListener';
```
Comment (jakubcolony) [inline src/handlers/colonies/colonyAdded.ts:1]: This is unchanged

File: src/handlers/colonies/colonyFundsClaimed.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { constants } from 'ethers';
```
Comment (jakubcolony) [inline src/handlers/colonies/colonyFundsClaimed.ts:1]: This is unchanged

File: src/handlers/colonies/colonyUpgraded.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { mutate } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/colonies/colonyUpgraded.ts:1]: This is unchanged

File: src/handlers/colonies/colonyVersionAdded.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { mutate } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/colonies/colonyVersionAdded.ts:1]: This is unchanged

File: src/handlers/colonies/transfer.ts [hunk 7]
```diff
@@ -0,0 +1,85 @@
+import { mutate, query } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/colonies/transfer.ts:1]: This is unchanged

File: src/handlers/extensions/extensionAddedToNetwork.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { ContractEvent } from '~/types';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionAddedToNetwork.ts:1]: This is unchanged

File: src/handlers/extensions/extensionDeprecated.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { mutate } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionDeprecated.ts:1]: This is unchanged

File: src/handlers/extensions/extensionInitialised.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { mutate } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionInitialised.ts:1]: This is unchanged

File: src/handlers/extensions/extensionInstalled.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import { extensionSpecificEventsListener } from '~/eventListener';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionInstalled.ts:1]: This is unchanged

File: src/handlers/extensions/extensionUninstalled.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import { ContractEvent } from '~/types';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionUninstalled.ts:1]: This is unchanged

File: src/handlers/extensions/extensionUpgraded.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { mutate } from '~/amplifyClient';
```
Comment (jakubcolony) [inline src/handlers/extensions/extensionUpgraded.ts:1]: This is unchanged

**Review by rdig — APPROVED** [review]
All good here, I actually like that you spun out actions into their own modules/files as they were starting to become heavy to read

**Review by willm30 — APPROVED** [review]
Managed to get it going and everything looking as expected 👍 👍

File: tsconfig.json [hunk 14]
```diff
@@ -25,17 +25,19 @@
...
-    "module": "commonjs",                                /* Specify what module code is generated. */
+    "module": "commonjs" /* Specify what module code is generated. */,
     // "rootDir": "./",                                  /* Specify the root folder within your source files. */
-    "moduleResolution": "node",                          /* Specify how TypeScript looks up a file from a given module specifier. */
-    // "baseUrl": "./",                                  /* Specify the base directory to resolve non-relative module names. */
-    // "paths": {},                                      /* Specify a set of entries that re-map imports to additional lookup locations. */
+    "moduleResolution": "node" /* Specify how TypeScript looks up a file from a given module specifier. */,
+    "baseUrl": "." /* Specify the base directory to resolve non-relative module names. */,
+    "paths": {
+      "~/*": ["src/*"]
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply tsconfig.json:33]: Good point. I was initially thinking we'd add more paths as we go and have a generic `~/` in the meantime but I guess it makes sense to just make it `~*` instead.

**Pull request merged**